spec/system/log_in_system_spec.rb

```diff
@@ -10,9 +10,23 @@
     end
   end
 
+  describe "User with no roles" do
+    before do
+      create(:user, :no_roles, email: "no_role_user@example.com")
+    end
+
+    it "should not allow the user to log in" do
+      visit "/users/sign_in"
+      fill_in "user_email", with: "no_role_user@example.com"
+      fill_in "user_password", with: DEFAULT_USER_PASSWORD
+      find('input[name="commit"]').click
+      expect(page).to have_content("You need to sign in before continuing.")
```

Comment:
I don't love this error message but I could not figure out how to change it. 

I tried `throw(:warden, action: :invalid)` but that still gave the same error message.

Reply:
Not ideal, but this kind of code is super hairy. I'd timebox it and if you can't change it, I'm OK with it as is. @cielf for final decision here.